aten/src/ATen/templates/TensorBody.h

```diff
@@ -360,55 +360,97 @@ class TORCH_API Tensor {
   }
 
   /// Returns a `Tensor`'s layout. Defined in Type.h
-  Layout layout() const noexcept;
+  Layout layout() const noexcept {
+    return impl_->layout();
+  }
 
   /// Returns a `Tensor`'s dtype (`TypeMeta`). Defined in TensorMethods.cpp
-  caffe2::TypeMeta dtype() const noexcept;
+  caffe2::TypeMeta dtype() const noexcept {
+    return impl_->dtype();
+  }
 
   /// Returns a `Tensor`'s device.
   inline Device device() const {
     return impl_->device();
   }
 
   /// Returns a `Tensor`'s device index.
-  int64_t get_device() const;
+  int64_t get_device() const {
+    // NB: this is not a native function to avoid dispatching overhead.
+    return impl_->get_device();
+  }
+
 
   /// Returns if a `Tensor` has CPU backend.
-  bool is_cpu() const;
+  bool is_cpu() const {
```

Comment:
Is there any benefit to making this and the other is_*() functions `noexcept` in release mode? AFAICT they only throw in debug, due to a TORCH_INTERNAL_ASSERT_DEBUG_ONLY in `DispatchKeySet::has()`.

Reply:
I don't think it matters for inline functions -- the compiler can infer that on its own.